spec/formatting.md

```diff
@@ -23,11 +23,15 @@ are treated as identical.
 To resolve the value of a _variable_,
 its _name_ is used to identify either a local variable,
 or a variable defined elsewhere.
-If a local variable and an externally defined one use the same name,
-the local variable takes precedence.
 
-It is an error for a local variable definition to
-refer to a local variable that's defined after it in the message.
+It is an error for the right-hand side of a local variable declaration to
```

Comment:
the right and left hand sides have names in the ABNF. You should use those. In this case it should be something like:

```suggestion
The `expression` part of a `declaration` MUST NOT refer to a `variable` that is declared after it in the `message`.
```

(Technically we could just say "an `expression` MUST NOT refer to a `variable` that has not been assigned in a previous `declaration` or passed in...")

Reply:
+1 to Addison's phrasing.

Reply:
I made an attempt to rephrase this. 

This would be much easier to specify if, in the grammar, `declaration` and `operand` were changed:

```
declaration = let s variable-definition [s] "=" [s] expression
operand = literal / variable-reference
option = name [s] "=" [s] operand
variable-definition = variable
variable-reference = variable
```

If this change were made, then I'd be able to write something like:

"The _variable_ of any `variable-reference` must be preceded by a full _declaration_ whose `variable-definition` has the same _variable_."

and could avoid referring to the enclosing context of "the operand".